_inc/lib/core-api/wpcom-endpoints/memberships.php

```diff
@@ -78,6 +78,55 @@ public function register_routes() {
 				),
 			)
 		);
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/products',
+			array(
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_products' ),
+					'permission_callback' => array( $this, 'get_status_permission_check' ),
+					'args'                => array(
+						'type'     => array(
+							'type'              => 'string',
+							'required'          => true,
+							'validate_callback' => function( $param ) {
+								return in_array( $param, array( 'donation' ), true );
+							},
+						),
+						'currency' => array(
+							'type'              => 'string',
+							'required'          => false,
+							'validate_callback' => function( $param ) {
+								return in_array(
+									$param,
+									array(
```

Comment:
I don't know if that's ideal. We already have a few of those lists:
https://github.com/Automattic/jetpack/blob/7f3bd724c184747db76937e367d1840c7a91fb6f/extensions/blocks/simple-payments/constants.js#L17
https://github.com/Automattic/jetpack/blob/f02d252966e432088b4c3bcc76c95729b5a885a6/modules/simple-payments/simple-payments.php#L392
https://github.com/Automattic/jetpack/blob/38bbcb7bbdd7da82d821cfb98751056d9eaf305e/modules/widgets/simple-payments.php#L31

Keeping them all up to date is going to prove more and more difficult. Maybe now would be a good time to switch to querying that list from an API endpoint?

If not now, do you think you could add a comment mentioning that this should be kept up to date, like we've done for the other lists?

Reply:
This list is actually different because Memberships are built on top of Stripe while Simple Payments uses PayPal. But you're right though, since this list of Stripe currencies is duplicated in the Recurring Payments block, so I added a comment in both files to remind folks it should be synced.

I love your idea of having a centralized place for retrieving the currencies like an endpoint, but I rather prefer to leave that for a follow-up since the scope is a bit bigger than this PR (i.e. the Premium Content block in FSE).